app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/LocalFileTransferActivity.java

```diff
@@ -0,0 +1,539 @@
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import androidx.fragment.app.DialogFragment;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentTransaction;
+
+import org.kiwix.kiwixmobile.KiwixApplication;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.utils.SharedPreferenceUtil;
+
+import java.util.ArrayList;
+
+import javax.inject.Inject;
+
+/**
+ * Created by @Aditya-Sood as a part of GSoC 2019.
+ *
+ * This activity is the starting point for the module used for sharing zims between devices.
+ *
+ * The module is used for transferring ZIM files from one device to another, from within the
+ * app. Two devices are connected to each other using WiFi Direct, followed by file transfer.
+ *
+ * The module uses this activity along with {@link DeviceListFragment} to manage connection
+ * and file transfer between the devices.
+ * */
+public class LocalFileTransferActivity extends AppCompatActivity implements WifiP2pManager.ChannelListener, DeviceListFragment.DeviceActionListener {
+
+  public static final String TAG = "LocalFileTransferActvty"; // Not a typo, 'Log' tags have a length upper limit of 25 characters
+  public static final int REQUEST_ENABLE_WIFI_P2P = 1;
+  public static final int REQUEST_ENABLE_LOCATION_SERVICES = 2;
+  private static final int PERMISSION_REQUEST_CODE_COARSE_LOCATION = 1;
+  private static final int PERMISSION_REQUEST_CODE_STORAGE_WRITE_ACCESS = 2;
+
+  @Inject
+  SharedPreferenceUtil sharedPreferenceUtil;
+
+  private ArrayList<Uri> fileUriArrayList;  // For sender device, stores Uris of files to be transferred
+  private Boolean fileSendingDevice = false;// Whether the device is the file sender or not
+
+
+  /* Variables related to the WiFi P2P API */
+  private boolean wifiP2pEnabled = false; // Whether WiFi has been enabled or not
+  private boolean retryChannel = false;   // Whether channel has retried connecting previously
+
+  private WifiP2pManager manager;         // Overall manager of Wifi p2p connections for the module
+  private WifiP2pManager.Channel channel; // Connects the module to device's underlying Wifi p2p framework
+
+  private final IntentFilter intentFilter = new IntentFilter(); // For specifying broadcasts (of the P2P API) that the module needs to respond to
+  private BroadcastReceiver receiver = null; // For receiving the broadcasts given by above filter
+
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_local_file_transfer);
+    setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT); // Protect AsyncTask from orientation changes
```

Comment:
do this in the manifest, not in code

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/db94a372b71aaa3f36cd11f3cf485b0cc6f28dab)